File: src/features/typing/data/go/subarray_sum_equals_k.go
package main

func subarraySum(nums []int, k int) int {
	count := make(map[int]int)
	count[0] = 1
	presum, total := 0, 0
	for _, num := range nums {
		presum += num
		total += count[presum-k]
		count[presum]++
	}
	return total
}

File: src/features/typing/data/ts/subarray_sum_equals_k.ts
function subarraySum(nums: number[], k: number): number {
    const count = new Map<number, number>();
    count.set(0, 1);
    let presum = 0;
    let total = 0;
    for (const num of nums) {
        presum += num;
        total += count.get(presum - k) ?? 0;
        count.set(presum, (count.get(presum) ?? 0) + 1);
    }
    return total;
}
